src/assisted_installer_controller/reboots_notifier.go

```diff
@@ -0,0 +1,106 @@
+package assisted_installer_controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-installer/src/common"
+	"github.com/openshift/assisted-installer/src/inventory_client"
+	"github.com/openshift/assisted-installer/src/ops"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+)
+
+const (
+	eventName            = "reboots_for_node"
+	eventMessageTemplate = "Node %s has been rebooted %d times before completing installation"
+)
+
+//go:generate mockgen -source=reboots_notifier.go -package=assisted_installer_controller -destination=mock_reboots_notifier.go
+type RebootsNotifier interface {
+	Start(ctx context.Context, nodeName string, hostId, infraenvId, clusterId *strfmt.UUID)
+	Finalize()
+}
+
+type rebootsNotifier struct {
+	cancelers      []context.CancelFunc
+	log            logrus.FieldLogger
+	wg             sync.WaitGroup
+	kubeconfigPath string
+	ops            ops.Ops
+	ic             inventory_client.InventoryClient
+	mu             sync.Mutex
+}
+
+func NewRebootsNotifier(ops ops.Ops, ic inventory_client.InventoryClient, log logrus.FieldLogger) RebootsNotifier {
+	return &rebootsNotifier{
+		log: log,
+		ops: ops,
+		ic:  ic,
+	}
+}
+
+func (r *rebootsNotifier) getKubeconfigPath(ctx context.Context) (string, error) {
+	if r.kubeconfigPath != "" {
+		return r.kubeconfigPath, nil
+	}
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if r.kubeconfigPath == "" {
+		var err error
+		if r.kubeconfigPath, err = common.DownloadKubeconfigNoingress(ctx, os.TempDir(), r.ic, r.log); err != nil {
```

Comment:
Asking, not urgent. We do it at couple of places, maybe we can download it once to known path and use it from there? as you do it here?

Reply:
I yes, I don't think it should be part of this PR